diff --git a/node_modules/spdy/lib/spdy/agent.js b/node_modules/spdy/lib/spdy/agent.js
index 4adc524e..a7ebaf4f 100644
--- a/node_modules/spdy/lib/spdy/agent.js
+++ b/node_modules/spdy/lib/spdy/agent.js
@@ -8,10 +8,23 @@ var util = require('util')
 var transport = require('spdy-transport')
 var debug = require('debug')('spdy:client')
 
-// Node.js 0.10 and 0.12 support
-Object.assign = process.versions.modules >= 46
-  ? Object.assign // eslint-disable-next-line
-  : util._extend
+var assign = Object.assign || function assign (target) {
+  if (target == null) {
+    target = {}
+  }
+  for (var i = 1; i < arguments.length; i++) {
+    var source = arguments[i]
+    if (!source) {
+      continue
+    }
+    var keys = Object.keys(source)
+    for (var j = 0; j < keys.length; j++) {
+      var key = keys[j]
+      target[key] = source[key]
+    }
+  }
+  return target
+}
 
 var EventEmitter = require('events').EventEmitter
 
@@ -104,7 +117,7 @@ proto._connect = function _connect (options, callback) {
   ]
 
   // TODO(indutny): reconnect automatically?
-  var socket = this.createConnection(Object.assign({
+  var socket = this.createConnection(assign({
     NPNProtocols: protocols,
     ALPNProtocols: protocols,
     servername: options.servername || options.host
@@ -139,7 +152,7 @@ proto._connect = function _connect (options, callback) {
     }
 
     debug('connected protocol=%j', protocol)
-    var connection = transport.connection.create(socket, Object.assign({
+    var connection = transport.connection.create(socket, assign({
       protocol: /spdy/.test(protocol) ? 'spdy' : 'http2',
       isServer: false
     }, state.options.connection || {}))
diff --git a/node_modules/spdy/lib/spdy/server.js b/node_modules/spdy/lib/spdy/server.js
index 1889fcd2..e76ba08b 100644
--- a/node_modules/spdy/lib/spdy/server.js
+++ b/node_modules/spdy/lib/spdy/server.js
@@ -11,10 +11,23 @@ var transport = require('spdy-transport')
 var debug = require('debug')('spdy:server')
 var EventEmitter = require('events').EventEmitter
 
-// Node.js 0.8, 0.10 and 0.12 support
-Object.assign = process.versions.modules >= 46
-  ? Object.assign // eslint-disable-next-line
-  : util._extend
+var assign = Object.assign || function assign (target) {
+  if (target == null) {
+    target = {}
+  }
+  for (var i = 1; i < arguments.length; i++) {
+    var source = arguments[i]
+    if (!source) {
+      continue
+    }
+    var keys = Object.keys(source)
+    for (var j = 0; j < keys.length; j++) {
+      var key = keys[j]
+      target[key] = source[key]
+    }
+  }
+  return target
+}
 
 var spdy = require('../spdy')
 
@@ -49,7 +62,7 @@ proto._init = function _init (base, options, handler) {
     'http/1.1', 'http/1.0'
   ]
 
-  var actualOptions = Object.assign({
+  var actualOptions = assign({
     NPNProtocols: protocols,
 
     // Future-proof
@@ -112,7 +125,7 @@ proto._handleConnection = function _handleConnection (socket, protocol) {
 
   socket.setNoDelay(true)
 
-  var connection = transport.connection.create(socket, Object.assign({
+  var connection = transport.connection.create(socket, assign({
     protocol: /spdy/.test(protocol) ? 'spdy' : 'http2',
     isServer: true
   }, state.options.connection || {}))
