diff --git a/node_modules/spdy-transport/lib/spdy-transport/utils.js b/node_modules/spdy-transport/lib/spdy-transport/utils.js
index 7f379f25..d8401050 100644
--- a/node_modules/spdy-transport/lib/spdy-transport/utils.js
+++ b/node_modules/spdy-transport/lib/spdy-transport/utils.js
@@ -3,10 +3,28 @@
 var util = require('util')
 var isNode = require('detect-node')
 
+var assignFallback = function assignFallback (target) {
+  if (target == null) {
+    target = {}
+  }
+  for (var i = 1; i < arguments.length; i++) {
+    var source = arguments[i]
+    if (!source) {
+      continue
+    }
+    var keys = Object.keys(source)
+    for (var j = 0; j < keys.length; j++) {
+      var key = keys[j]
+      target[key] = source[key]
+    }
+  }
+  return target
+}
+
 // Node.js 0.8, 0.10 and 0.12 support
-Object.assign = (process.versions.modules >= 46 || !isNode)
-  ? Object.assign // eslint-disable-next-line
-  : util._extend
+if (!Object.assign && isNode) {
+  Object.assign = assignFallback
+}
 
 function QueueItem () {
   this.prev = null
